src/Domain.LinnApps/ImportBooks/ImportBookService.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace Linn.Stores.Domain.LinnApps.ImportBooks
+{
+    using System.Collections.Generic;
+    using Linn.Common.Persistence;
+    using Linn.Stores.Domain.LinnApps.Parts;
+
+    public class ImportBookService : IImportBookService
+    {
+        private readonly IRepository<ImportBookInvoiceDetail, ImportBookInvoiceDetailKey> InvoiceDetailRepository;
+        private readonly IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> OrderDetailRepository;
+        private readonly IRepository<ImportBookPostEntry, ImportBookPostEntryKey> PostEntryRepository;
+
+
+        public ImportBookService(
+            IRepository<ImportBookInvoiceDetail, ImportBookInvoiceDetailKey> invoiceDetailRepository,
+            IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> orderDetailRepository,
+            IRepository<ImportBookPostEntry, ImportBookPostEntryKey> postEntryRepository)
+        {
+            this.InvoiceDetailRepository = invoiceDetailRepository;
+            this.OrderDetailRepository = orderDetailRepository;
+            this.PostEntryRepository = postEntryRepository;
+        }
+
+        public void Update(
+            ImportBook from,
+            IEnumerable<ImportBookInvoiceDetail> invoiceDetails,
+            IEnumerable<ImportBookOrderDetail> orderDetails,
+            IEnumerable<ImportBookPostEntry> postEntries)
+        {
+            this.UpdateInvoiceDetails(invoiceDetails);
+
+            this.UpdateOrderDetails(orderDetails);
+
+            this.UpdatePostEntries(postEntries);
+        }
+
+        public void UpdateInvoiceDetails(IEnumerable<ImportBookInvoiceDetail> details)
+        {
+            foreach (var detail in details)
+            {
+                var currentDetail = this.InvoiceDetailRepository.FindById(new ImportBookInvoiceDetailKey(detail.ImportBookId, detail.LineNumber));
+
+                if (currentDetail == null)
+                {
+                    this.CreateInvoiceDetail(detail);
+                }
+                else
+                {
+                    currentDetail.InvoiceNumber = detail.InvoiceNumber;
+                    currentDetail.InvoiceValue = detail.InvoiceValue;
+                }
+            }
+        }
+
+        public void UpdateOrderDetails(IEnumerable<ImportBookOrderDetail> details)
+        {
+            foreach (var detail in details)
+            {
+                var currentDetail = this.OrderDetailRepository.FindById(new ImportBookOrderDetailKey(detail.ImportBookId, detail.LineNumber));
```

Comment:
as above. The child objects should never have their own repository ideally.

Reply:
I think this is what I was confused about at the start of this - the child objects are from different tables, so should I be doing a link in the service db context like importBook hasMany importBookInvoiceDetails sort of thing? If not how do I get them without their own repo?

Reply:
Yeah sorry. i was assuming all the children are linked to import book through ef in servicedbcontext

Reply:
And returned in the repo